assets/js/blocks/checkout/inner-blocks/checkout-collection-method-block/shared/rate-price.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { getSetting } from '@woocommerce/settings';
+import { createInterpolateElement } from '@wordpress/element';
+import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
+import FormattedMonetaryAmount from '@woocommerce/base-components/formatted-monetary-amount';
+import type { CartShippingPackageShippingRate } from '@woocommerce/type-defs/cart';
+
+export const RatePrice = ( {
+	minRate,
+	maxRate,
+}: {
+	minRate: CartShippingPackageShippingRate | undefined;
+	maxRate: CartShippingPackageShippingRate | undefined;
+} ) => {
+	if ( ! minRate && ! maxRate ) {
+		return null;
+	}
+	const minRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
+		? parseInt( minRate.price, 10 ) + parseInt( minRate.taxes, 10 )
+		: parseInt( minRate.price, 10 );
+	const maxRatePrice = getSetting( 'displayCartPricesIncludingTax', false )
+		? parseInt( maxRate.price, 10 ) + parseInt( maxRate.taxes, 10 )
+		: parseInt( maxRate.price, 10 );
+
+	return (
+		<span className="wc-block-checkout__collection-item-price">
+			{ minRatePrice === 0 && maxRatePrice === 0 ? (
+				<em>{ __( 'free', 'woo-gutenberg-products-block' ) }</em>
+			) : (
+				createInterpolateElement(
+					__( 'from <price />', 'woo-gutenberg-products-block' ),
+					{
+						price:
+							minRatePrice === 0 ? (
+								<em>
+									{ __(
+										'free',
+										'woo-gutenberg-products-block'
+									) }
+								</em>
+							) : (
+								<FormattedMonetaryAmount
+									currency={ getCurrencyFromPriceResponse(
+										minRate
+									) }
+									value={ minRatePrice }
+								/>
+							),
+					}
+				)
+			) }
```

Comment:
This still causes the UI to show `from free` which is awkward. Is it expected?

<img width="506" alt="image" src="https://user-images.githubusercontent.com/6165348/195099395-bd7aa881-ec4d-4138-9871-26da4d1bb46a.png">

Reply:
Yes. Before I introduced `maxRatePrice`, if there were 2 Local Pickups, one paid, one free, the text would just read "free" which is misleading. Free is the lowest price.

Reply:
Translation wise this is a bit awkward because now we separate a static word "from" from another static word "free". How about doing it like this?

```
<span className="wc-block-checkout__collection-item-price">
	{ minRatePrice === 0
		? maxRatePrice === 0 && (
				<em>
					{ __( 'free', 'woo-gutenberg-products-block' ) }
				</em>
			)
		: createInterpolateElement(
				__( 'from <price />', 'woo-gutenberg-products-block' ),
				{
					price: (
						<FormattedMonetaryAmount
							currency={ getCurrencyFromPriceResponse(
								minRate
							) }
							value={ minRatePrice }
						/>
					),
				}
			) }
</span>
```

Reply:
Yeah that's annoying. We'll have to think some more. I had to change your commit slightly because it made the full string in caps.